### ISSUE TITLE
com.hazelcast.jet.cdc.DebeziumCdcIntegrationTest

### ISSUE THREAD

**Issue by sumnerib** [hazelcast/hazelcast#26264]
_master_ (commit d200fd2a102eb0467ceebae4fb2cad95abd349c9)

Failed on CorrettoJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-CorrettoJDK17-arm64-nightly/4/testReport/junit/com.hazelcast.jet.cdc/DebeziumCdcIntegrationTest/

<details><summary>Stacktrace:</summary>

```
java.lang.NoClassDefFoundError: org/apache/commons/codec/Charsets
	at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.<init>(TarArchiveOutputStream.java:212)
	at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.<init>(TarArchiveOutputStream.java:157)
	at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.<init>(TarArchiveOutputStream.java:147)
	at org.testcontainers.containers.ContainerState.copyFileToContainer(ContainerState.java:350)
	at org.testcontainers.containers.ContainerState.copyFileToContainer(ContainerState.java:331)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:444)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:357)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:347)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:333)
	at com.hazelcast.jet.cdc.DebeziumCdcIntegrationTest.mysql(DebeziumCdcIntegrationTest.java:78)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.codec.Charsets
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 17 more

```
</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: noFailWhenBeforeIsNotPresent in 50.053 seconds.
Started Running Test: mysql
01:28:42,152 DEBUG |mysql| - [PrefixingImageNameSubstitutor] mysql - No prefix is configured
01:28:42,152 DEBUG |mysql| - [ImageNameSubstitutor] mysql - Did not find a substitute image for debezium/example-mysql:2.3.0.Final (using image substitutor: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor'))
01:28:42,154 DEBUG |mysql| - [AbstractImagePullPolicy] mysql - Not available locally, should pull image: debezium/example-mysql:2.3.0.Final
01:28:42,154  INFO |mysql| - [Final] mysql - Pulling docker image: debezium/example-mysql:2.3.0.Final. Please be patient; this may take some time but only needs to be done once.
01:28:42,154 DEBUG |mysql| - [RegistryAuthLocator] mysql - Looking up auth config for image: debezium/example-mysql:latest at registry: https://index.docker.io/v1/
01:28:42,154 DEBUG |mysql| - [RegistryAuthLocator] mysql - No matching Auth Configs - falling back to defaultAuthConfig [null]
01:28:42,154 DEBUG |mysql| - [AuthDelegatingDockerClientConfig] mysql - Effective auth config [null]
01:28:42,338  INFO |mysql| - [Final] docker-java-stream--630443632 - Starting to pull image
01:28:42,340  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
01:28:42,536  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers: 12 pending,  1 downloaded,  0 extracted, (17 KB/? MB)
01:28:42,626  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers: 11 pending,  2 downloaded,  0 extracted, (507 KB/? MB)
01:28:42,718  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers: 10 pending,  3 downloaded,  0 extracted, (3 MB/? MB)
01:28:42,720  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  9 pending,  4 downloaded,  0 extracted, (3 MB/? MB)
01:28:42,833  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  8 pending,  5 downloaded,  0 extracted, (3 MB/? MB)
01:28:43,880  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  7 pending,  6 downloaded,  0 extracted, (118 MB/? MB)
01:28:43,976  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  6 pending,  7 downloaded,  0 extracted, (136 MB/? MB)
01:28:44,011  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  5 pending,  8 downloaded,  0 extracted, (136 MB/? MB)
01:28:44,032  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  4 pending,  9 downloaded,  0 extracted, (136 MB/? MB)
01:28:44,085  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  3 pending, 10 downloaded,  0 extracted, (143 MB/? MB)
01:28:44,131  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  2 pending, 11 downloaded,  0 extracted, (143 MB/? MB)
01:28:44,322  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  0 extracted, (146 MB/? MB)
01:28:44,644  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  1 extracted, (147 MB/? MB)
01:28:44,689  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  2 extracted, (147 MB/? MB)
01:28:44,745  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  3 extracted, (148 MB/? MB)
01:28:44,869  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  4 extracted, (149 MB/? MB)
01:28:44,914  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  5 extracted, (149 MB/? MB)
01:28:44,957  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  6 extracted, (149 MB/? MB)
01:28:45,702  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  7 extracted, (151 MB/? MB)
01:28:45,743  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  8 extracted, (151 MB/? MB)
01:28:47,717  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded,  9 extracted, (162 MB/? MB)
01:28:47,764  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded, 10 extracted, (162 MB/? MB)
01:28:47,808  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded, 11 extracted, (162 MB/? MB)
01:28:47,858  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded, 12 extracted, (162 MB/? MB)
01:28:47,902  INFO |mysql| - [Final] docker-java-stream--630443632 - Pulling image layers:  1 pending, 12 downloaded, 13 extracted, (162 MB/? MB)
01:28:47,920  INFO |mysql| - [Final] docker-java-stream--630443632 - Pull complete. 13 layers, pulled in 5s (downloaded 162 MB at 32 MB/s)
01:28:47,920  INFO |mysql| - [Final] mysql - Image debezium/example-mysql:2.3.0.Final pull took PT5.765635285S
01:28:47,923  INFO |mysql| - [Final] mysql - Creating container for image: debezium/example-mysql:2.3.0.Final
01:28:47,923 DEBUG |mysql| - [RegistryAuthLocator] mysql - Looking up auth config for image: debezium/example-mysql:2.3.0.Final at registry: https://index.docker.io/v1/
01:28:47,923 DEBUG |mysql| - [RegistryAuthLocator] mysql - No matching Auth Configs - falling back to defaultAuthConfig [null]
01:28:47,923 DEBUG |mysql| - [AuthDelegatingDockerClientConfig] mysql - Effective auth config [null]
01:28:51,791 DEBUG |mysql| - [MountableFile] mysql - Copying classpath resource(s) from jar:file:/home/ec2-user/.m2/repository/org/testcontainers/mysql/1.19.5/mysql-1.19.5.jar!/mysql-default-conf to /tmp/.testcontainers-tmp-14605696370472643034 to permit Docker to bind
01:28:51,791 DEBUG |mysql| - [MountableFile] mysql - Copying resource mysql-default-conf from JAR file /home/ec2-user/.m2/repository/org/testcontainers/mysql/1.19.5/mysql-1.19.5.jar
01:28:51,791 DEBUG |mysql| - [MountableFile] mysql - Copying classpath resource(s) from jar:file:/home/ec2-user/.m2/repository/org/testcontainers/mysql/1.19.5/mysql-1.19.5.jar!/mysql-default-conf to /tmp/.testcontainers-tmp-14605696370472643034 to permit Docker to bind
01:28:51,792 DEBUG |mysql| - [MountableFile] mysql - Copying resource mysql-default-conf from JAR file /home/ec2-user/.m2/repository/org/testcontainers/mysql/1.19.5/mysql-1.19.5.jar
01:28:51,808 DEBUG |mysql| - [ResourceReaper] mysql - Removed container and associated volume(s): debezium/example-mysql:2.3.0.Final
01:28:51,837  INFO |mysql| - [HazelcastTestSupport] Time-limited test - Shutting down node factory as @After action
BuildInfo right after mysql(com.hazelcast.jet.cdc.DebeziumCdcIntegrationTest): BuildInfo{version='5.4.0-SNAPSHOT', build='20240221', buildNumber=20240221, revision=, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'mysql(com.hazelcast.jet.cdc.DebeziumCdcIntegrationTest):'
01:28:40, accumulated pauses: 23 ms, max pause: 1 ms, pauses over 1000 ms: 0
01:28:45, accumulated pauses: 22 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:28:50, accumulated pauses: 31 ms, max pause: 24 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-CorrettoJDK17-arm64-nightly/4/

**Comment by JackPGreen** [comment]
Caused by https://github.com/hazelcast/hazelcast-mono/pull/841

[New `commons-compress` now declares the `codec` dependency optional](https://github.com/apache/commons-compress/pull/482).

**Comment by AleksPeychev** [comment]
Failed again on _master_ with oracle-17 (little-endian): https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-nightly-little-endian/67/testReport/junit/com.hazelcast.jet.cdc/DebeziumCdcIntegrationTest/

**Comment by JackPGreen** [comment]
> Failed again on _master_ with
> 
> * oracle-17 (little-endian) https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-nightly-little-endian/67/testReport/junit/com.hazelcast.jet.cdc/DebeziumCdcIntegrationTest/
> * zing-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17-nightly/66/testReport/junit/com.hazelcast.jet.cdc/DebeziumCdcIntegrationTest/

Can't reproduce locally.

From the Jenkins logs:
> Checking out Revision ab28c3d381cbdfd43f9f943c2daada10bd29b02a (origin/master)

[ab28c3d381cbdfd43f9f943c2daada10bd29b02a](https://github.com/hazelcast/hazelcast-mono/commit/e611f82c6cd8afa5764d5ce415585a6a5085eb39) was before https://github.com/hazelcast/hazelcast-mono/pull/894 was merged ([e611f82c6cd8afa5764d5ce415585a6a5085eb39](https://github.com/hazelcast/hazelcast-mono/commit/e611f82c6cd8afa5764d5ce415585a6a5085eb39))
